modules/plottinggl/src/processors/colorscalelegend.cpp

```diff
@@ -221,10 +221,16 @@ std::tuple<ivec2, ivec2, ivec2, ivec2> ColorScaleLegend::getPositions(ivec2 dime
 }
 
 void ColorScaleLegend::process() {
-    utilgl::activateTargetAndClearOrCopySource(outport_, inport_);
     if (!enabled_) {
+        if (inport_.isReady()) {
```

Comment:
isConnected? 
If it is optional it will be ready even if not connected

Reply:
Then you need to question all places we use this test. For example here
https://github.com/inviwo/inviwo/blob/1f5e9dfff6b8a62d6079a47ddc47f1816e79d02a/modules/opengl/src/texture/textureutils.cpp#L114-L131

Reply:
And, if the port is connected but not ready, what should happen then?

Reply:
You are right, the processor will be ready but not the optional port itself (looked at the documentation).